dmt/main/mixins.py

```diff
@@ -0,0 +1,74 @@
+import pytz
+from datetime import date, datetime
+from dateutil.relativedelta import relativedelta
+from django.conf import settings
+from django.utils.dateparse import parse_date
+
+
+class DaterangeMixin(object):
+    """A mixin for views that use daterange-datepicker pagination.
+
+    This is meant to be attached to a View. The mixin calculates dates
+    using two keyword arguments in the GET request:
+
+    - interval_start
+    - interval_end
+    """
+
+    interval_start = None
+    interval_end = None
+    _today = None
+
+    def today(self):
+        if not self._today:
+            self._today = date.today()
+        return self._today
+
+    def calc_interval(self):
+        # Calculate from the beginning of the first day in the range
+        # to the end of the last day.
+        naive_start = datetime.combine(
+            self.interval_start, datetime.min.time())
+        naive_end = datetime.combine(
+            self.interval_end, datetime.max.time())
+
+        # Convert to TZ-aware, based on the current timezone.
+        aware_start = pytz.timezone(settings.TIME_ZONE).localize(
+            naive_start, is_dst=None)
+        aware_end = pytz.timezone(settings.TIME_ZONE).localize(
+            naive_end, is_dst=None)
+
+        self.interval_start = aware_start
+        self.interval_end = aware_end
+
+    def get_params(self):
+        """Update the interval based on request params."""
+
+        self.interval_start = self.request.GET.get('interval_start', None)
+        self.interval_end = self.request.GET.get('interval_end', None)
+
+        if not self.interval_start:
+            self.interval_start = self.today() - relativedelta(months=1)
+        if not self.interval_end:
+            self.interval_end = self.today()
+
+        try:
+            self.interval_start = parse_date(self.interval_start)
+        except TypeError:
+            pass
+        try:
+            self.interval_end = parse_date(self.interval_end)
+        except TypeError:
+            pass
+
+        self.calc_interval()
```

Comment:
good point.. i'm going to move lines 50-53 after these try/excepts, that we there is no repeated code